crates/tests/src/differential.rs

```diff
@@ -0,0 +1,148 @@
+//! Tests that check for differences between Solidity and Fe implementations of similar contracts
+#![cfg(feature = "solc-backend")]
+use proptest::prelude::*;
+
+use fe_compiler_test_utils::*;
+use fe_compiler_test_utils::{self as test_utils};
+
+struct DualHarness {
+    fe_harness: ContractHarness,
+    solidity_harness: ContractHarness,
+}
+
+struct CaptureResult<'a> {
+    fe_capture: evm::Capture<(evm::ExitReason, Vec<u8>), std::convert::Infallible>,
+    solidity_capture: evm::Capture<(evm::ExitReason, Vec<u8>), std::convert::Infallible>,
+    name: &'a str,
+    input: &'a [ethabi::Token],
+}
+
+impl<'a> CaptureResult<'a> {
+    pub fn assert_perfomed_equal(&self) {
+        assert_eq!(
+            self.fe_capture, self.solidity_capture,
+            "Called {} with input: {:?}",
+            self.name, self.input
+        )
+    }
+
+    pub fn assert_reverted(&self) {
+        if !matches!(
+            (self.fe_capture.clone(), self.solidity_capture.clone()),
+            (
+                evm::Capture::Exit((evm::ExitReason::Revert(_), _)),
+                evm::Capture::Exit((evm::ExitReason::Revert(_), _))
+            )
+        ) {
+            panic!(
+                "Asserted both revert but was: Fe: {:?} Solidity: {:?}",
+                self.fe_capture, self.solidity_capture
+            )
+        }
+    }
+
+    pub fn performed_equal(&self) -> bool {
+        self.fe_capture == self.solidity_capture
+    }
+}
+
+impl<'a> DualHarness {
+    pub fn from_fixture(
+        executor: &mut Executor,
+        fixture: &str,
+        contract_name: &str,
+        init_params: &[ethabi::Token],
+    ) -> DualHarness {
+        let fe_harness = test_utils::deploy_contract(
+            executor,
+            &format!("differential/{}.fe", fixture),
+            contract_name,
+            init_params,
+        );
+        let solidity_harness = test_utils::deploy_solidity_contract(
+            executor,
+            &format!("differential/{}.sol", fixture),
+            contract_name,
+            init_params,
+        );
+        DualHarness {
+            fe_harness,
+            solidity_harness,
+        }
+    }
+
+    pub fn capture_call(
+        &self,
+        executor: &mut Executor,
+        name: &'a str,
+        input: &'a [ethabi::Token],
+    ) -> CaptureResult<'a> {
+        let fe_capture = self.fe_harness.capture_call(executor, name, input);
+        let solidity_capture = self.solidity_harness.capture_call(executor, name, input);
+
+        CaptureResult {
+            fe_capture,
+            solidity_capture,
+            name,
+            input,
+        }
+    }
+}
+
+proptest! {
+
+    #[test]
+    fn math_u8(val in 0u8..=255, val2 in 0u8..=255) {
+        with_executor(&|mut executor| {
+
+            let harness = DualHarness::from_fixture(&mut executor, "math_u8", "Foo", &[]);
```

Comment:
Yeah, this would only be feasible with contracts that don't mutate state. So if we end up running alot of tests that meet this criteria, then it might be worth looking into.

It would be nice to know how long it takes to initialize the executor and harness vs how long it takes to run the tests..